frontend/src/LineGraph.res

```diff
@@ -336,30 +346,58 @@ let make = React.memo((
     let stdDev = computeStdDev(~mean, values)
     DataRow.valueWithErrorBars(~mid=mean, ~low=mean -. stdDev, ~high=mean +. stdDev)
   }
-  let constantSeries = dataSet->Belt.Array.map(computeConstantSeries)
 
   // Dygraph needs null values, and cannot handle NaNs correctly
   let convertNanToNull = (xs: DataRow.t) =>
     Belt.Array.map(xs, x => Js.Float.isNaN(x) ? Obj.magic(Js.null) : x)
 
-  let nullConstantSeries = constantSeries->Belt.Array.map(convertNanToNull)
+  let graphColors = labels => {
+    let labels_ = labels->Belt.Option.getWithDefault([])
+    let n = category10colors->Belt.Array.length
+    let colorsArray =
+      n > labels_->Belt.Array.length
+        ? category10colors
+        : labels_->Belt.Array.mapWithIndex((idx, _) =>
+            category10colors->Belt.Array.getExn(mod(idx, n))
+          )
+    colorsArray->Belt.Array.keepWithIndex((_, idx) =>
+      linesState->Belt.Array.get(idx)->Belt.Option.getWithDefault(true)
+    )
+  }
 
-  let originalLabels = labels
-  let labels = labels->Belt.Option.map(labels => {
-    let means = Belt.Array.length(labels) > 1 ? labels->Belt.Array.map(x => "mean:" ++ x) : ["mean"]
-    Belt.Array.concatMany([["idx"], labels, means])
-  })
+  let makeDygraphLabels = labels =>
+    labels
+    ->Belt.Option.getWithDefault([])
+    ->Belt.Array.keepWithIndex((_, idx) =>
+      linesState->Belt.Array.get(idx)->Belt.Option.getWithDefault(true)
+    )
+    ->(
+      labels => {
+        let means =
+          Belt.Array.length(labels) > 1 ? labels->Belt.Array.map(x => "mean:" ++ x) : ["mean"]
+        Belt.Array.concatMany([["idx"], labels, means])
+      }
+    )
 
   let makeDygraphData = (data: array<DataRow.row>) => {
-    // Dygraph does not display the last tick, so a dummy value
-    // is added a the end of the data to overcome this.
-    // See: https://github.com/danvk/dygraphs/issues/506
+    let data =
+      data->Belt.Array.keepWithIndex((_, idx) =>
+        linesState->Belt.Array.get(idx)->Belt.Option.getWithDefault(true)
+      )
+    let constantSeries = data->Belt.Array.map(computeConstantSeries)
+    let nullConstantSeries = constantSeries->Belt.Array.map(convertNanToNull)
+
     let data =
       data
+      // Dygraph does not display the last tick, so a dummy value
+      // is added a the end of the data to overcome this.
+      // See: https://github.com/danvk/dygraphs/issues/506
       ->Belt.Array.map(x => Belt.Array.concat(x, [DataRow.dummyValue]))
       ->Belt.Array.map(x => x->Belt.Array.map(convertNanToNull))
 
-    let n = (data->Belt.Array.map(Belt.Array.length))[0]
+    let n =
+      data->Belt.Array.map(Belt.Array.length)->Belt.Array.get(0)->Belt.Option.getWithDefault(0)
```

Comment:
The `length` could be done as a last step, to skip the `map` :)

Reply:
Ah, right. Thanks!